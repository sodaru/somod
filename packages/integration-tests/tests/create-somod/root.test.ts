/**
 * @jest-environment steps
 */

import { existsSync } from "fs";
import { unlink } from "fs/promises";
import {
  createFiles,
  createTempDir,
  deleteDir,
  readFiles
} from "nodejs-file-utils";
import { join } from "path";
import { execute } from "../utils";

describe("Test the create-somod", () => {
  let dir: string;
  let somodDir: string;

  beforeAll(async () => {
    dir = createTempDir("test-somod-create");
    createFiles(dir, {
      ".npmrc": "registry=http://localhost:8000\ncache=./.npm"
    });
  });

  afterEach(() => {
    if (somodDir && existsSync(somodDir)) {
      deleteDir(somodDir);
    }
  });

  afterAll(() => {
    deleteDir(dir);
  });

  test("help", async () => {
    const result = await execute(
      dir,
      "npx",
      ["create-somod", "-h"],
      { return: "on", show: "off" },
      { return: "on", show: "off" }
    );
    expect(result.stdout).toMatchSnapshot();
  });

  test("without options", async () => {
    somodDir = join(dir, "my-module");
    const result = await execute(
      dir,
      "npx",
      ["create-somod"],
      { return: "on", show: "off" },
      { return: "on", show: "off" }
    );
    expect(result["failed"]).toBeUndefined();
    deleteDir(join(somodDir, "node_modules"));
    deleteDir(join(somodDir, ".git"));
    await unlink(join(somodDir, "package-lock.json"));
    expect(readFiles(somodDir)).toMatchSnapshot();
  }, 60000);

  test(
    "with module name",
    async () => {
      somodDir = join(dir, "new-somod-module");
      const result = await execute(
        dir,
        "npx",
        ["create-somod", "new-somod-module"],
        { return: "on", show: "off" },
        { return: "on", show: "off" }
      );
      expect(result["failed"]).toBeUndefined();
      deleteDir(join(somodDir, "node_modules"));
      deleteDir(join(somodDir, ".git"));
      await unlink(join(somodDir, "package-lock.json"));
      expect(readFiles(somodDir)).toMatchSnapshot();
    },
    2 * 60000
  );

  test(
    "with out git eslint and prettier",
    async () => {
      somodDir = join(dir, "my-module");
      const result = await execute(
        dir,
        "npx",
        ["create-somod", "--no-git", "--no-eslint", "--no-prettier"],
        { return: "on", show: "off" },
        { return: "on", show: "off" }
      );
      expect(result["failed"]).toBeUndefined();
      deleteDir(join(somodDir, "node_modules"));
      deleteDir(join(somodDir, ".git"));
      await unlink(join(somodDir, "package-lock.json"));
      expect(readFiles(somodDir)).toMatchSnapshot();
    },
    2 * 60000
  );
});
