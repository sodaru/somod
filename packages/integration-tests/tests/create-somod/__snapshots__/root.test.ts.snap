// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the create-somod help 1`] = `
"Usage: create-somod [options] [modName]

Arguments:
  modName             module Directory name to create project (default:
                      \\"my-module\\")

Options:
  -v, --verbose       enable verbose
  --ui                Initialize only UI
  --serverless        Initialize only Serverless
  --version           somod SDK Version
  --template-version  somod-template Version
  --no-git            Skip git initialization
  --no-prettier       Skip prettier initialization
  --no-eslint         Skip eslint initialization
  --no-files          Skip Sample files
  --npm-prompt        Prompt for input during npm init
  -h, --help          display help for command
"
`;

exports[`Test the create-somod with module name 1`] = `
Object {
  ".eslintignore": "node_modules
.somod
/build
/parameters.json
.aws-sam
samconfig.toml
/template.yaml
.next
tsconfig.json
/pages
/public
next-env.d.ts
.env
next.config.js",
  ".gitignore": "node_modules
.somod
/build
/parameters.json
.aws-sam
samconfig.toml
/template.yaml
.next
tsconfig.json
/pages
/public
next-env.d.ts
.env
next.config.js",
  ".prettierignore": "node_modules
.somod
/build
/parameters.json
.aws-sam
samconfig.toml
/template.yaml
.next
tsconfig.json
/pages
/public
next-env.d.ts
.env
next.config.js",
  "lib/index.ts": "export const Welcome = (): string => {
  return \\"Welcome to SOMOD Framework. learn more at https://somod.dev\\";
};
",
  "package.json": "{
  \\"name\\": \\"new-somod-module\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"scripts\\": {
    \\"prettier\\": \\"npx prettier --check --ignore-unknown --no-error-on-unmatched-pattern ./**/*\\",
    \\"eslint\\": \\"npx eslint ./ --no-error-on-unmatched-pattern\\",
    \\"prebuild\\": \\"npm run prettier && npm run eslint\\",
    \\"build\\": \\"npx somod build\\",
    \\"pretest\\": \\"npm run build\\",
    \\"test\\": \\"echo 'No Tests'\\",
    \\"prepack\\": \\"npm run test\\",
    \\"start\\": \\"npx somod start\\",
    \\"deploy\\": \\"npx somod deploy\\"
  },
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"ISC\\",
  \\"devDependencies\\": {
    \\"eslint-config-next\\": \\"^13.1.6\\",
    \\"eslint-config-sodaru\\": \\"^1.0.1\\",
    \\"prettier-config-sodaru\\": \\"^1.0.0\\",
    \\"somod\\": \\"^1.15.10\\"
  },
  \\"module\\": \\"build/lib/index.js\\",
  \\"typings\\": \\"build/lib/index.d.ts\\",
  \\"files\\": [
    \\"build\\"
  ],
  \\"sideEffects\\": false,
  \\"somod\\": \\"1.15.10\\"
}
",
  "parameters.yaml": "# yaml-language-server: $schema=./node_modules/somod-schema/schemas/parameters/index.json

parameters: {}
",
  "serverless/template.yaml": "# yaml-language-server: $schema=../node_modules/somod-schema/schemas/serverless-template/index.json

Resources:
  SampleParameter:
    Type: AWS::SSM::Parameter
    Properties:
      Name: Hello
      Description: \\"TODO: This is a sample resource, Delete this and add the valid resources for your module\\"
      Type: String
      Value: Good Luck
",
  "tsconfig.somod.json": "{
  \\"compilerOptions\\": {
    \\"allowUmdGlobalAccess\\": true,
    \\"outDir\\": \\"build\\",
    \\"declaration\\": true,
    \\"target\\": \\"ES5\\",
    \\"module\\": \\"ESNext\\",
    \\"rootDir\\": \\"./\\",
    \\"lib\\": [\\"ESNext\\", \\"DOM\\", \\"DOM.Iterable\\"],
    \\"moduleResolution\\": \\"Node\\",
    \\"esModuleInterop\\": true,
    \\"importHelpers\\": true,
    \\"skipLibCheck\\": true,
    \\"jsx\\": \\"react-jsx\\"
  },
  \\"include\\": [\\"lib\\", \\"ui\\", \\"serverless\\"],
  \\"exclude\\": []
}
",
  "ui/config.yaml": "# yaml-language-server: $schema=../node_modules/somod-schema/schemas/ui-config/index.json

env: {}
imageDomains: []
publicRuntimeConfig: {}
serverRuntimeConfig: {}
",
  "ui/pages/_document.tsx": "import { Html, Head, Main, NextScript } from \\"next/document\\";

const Document = () => {
  return (
    <Html>
      <Head>
        <link
          href=\\"https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@300;400;500;700&family=Roboto:wght@300;400;500;700&display=swap\\"
          rel=\\"stylesheet\\"
        />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
};

export default Document;
",
  "ui/pages/index.tsx": "import Head from \\"next/head\\";

export default function Home() {
  return (
    <div className=\\"container\\">
      <Head>
        <title key=\\"title\\">Create Somod Module</title>
        <link key=\\"favicon\\" rel=\\"icon\\" href=\\"/favicon.ico\\" />
      </Head>

      <main>
        <h1 className=\\"title\\">
          Welcome to <a href=\\"https://somod.dev\\">SOMOD!</a>
        </h1>

        <p className=\\"description\\">
          Get started by editing <code>ui/pages/index.tsx</code>
        </p>

        <div className=\\"grid\\">
          <a href=\\"https://docs.somod.dev\\" className=\\"card\\">
            <h3>Documentation &rarr;</h3>
            <p>Find in-depth information about SOMOD features and API.</p>
          </a>

          <a
            href=\\"https://www.youtube.com/channel/UCl6JNhHf9T7O1mxgnwwXvLw\\"
            className=\\"card\\"
          >
            <h3>Learn &rarr;</h3>
            <p>Learn SOMOD framework from these Explainer videos</p>
          </a>

          <a href=\\"https://github.com/somod-dev\\" className=\\"card\\">
            <h3>Examples &rarr;</h3>
            <p>Discover and deploy boilerplate example SOMOD projects.</p>
          </a>
        </div>
      </main>

      <footer>
        <a href=\\"https://somod.dev\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
          Powered by{\\" \\"}
          <svg
            width=\\"24\\"
            height=\\"24\\"
            viewBox=\\"0 0 100 100\\"
            version=\\"1.1\\"
            xmlns=\\"http://www.w3.org/2000/svg\\"
          >
            <polygon
              fill=\\"#ffb476\\"
              points=\\"0,0 100,0 100,20 40,20 40,50 0,50\\"
            ></polygon>
            <polygon
              fill=\\"#004b89\\"
              points=\\"0,80 60,80 60,50 100,50 100,100 0,100\\"
            ></polygon>
          </svg>
        </a>
      </footer>

      <style>{\`
        .container {
          min-height: 100vh;
          padding: 0 0.5rem;
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
        }

        main {
          padding: 5rem 0;
          flex: 1;
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
        }

        footer {
          width: 100%;
          height: 100px;
          border-top: 1px solid #eaeaea;
          display: flex;
          justify-content: center;
          align-items: center;
        }

        footer svg {
          margin-left: 0.5rem;
        }

        footer a {
          display: flex;
          justify-content: center;
          align-items: center;
        }

        a {
          color: inherit;
          text-decoration: none;
        }

        .title a {
          color: #004b89;
          text-decoration: none;
        }

        .title a:hover,
        .title a:focus,
        .title a:active {
          text-decoration: underline;
        }

        .title {
          margin: 0;
          line-height: 1.15;
          font-size: 4rem;
        }

        .title,
        .description {
          text-align: center;
        }

        .description {
          line-height: 1.5;
          font-size: 1.5rem;
        }

        code {
          background: #fafafa;
          border-radius: 5px;
          padding: 0.75rem;
          font-size: 1.1rem;
          font-family: Menlo, Monaco, Lucida Console, Liberation Mono,
            DejaVu Sans Mono, Bitstream Vera Sans Mono, Courier New, monospace;
        }

        .grid {
          display: flex;
          align-items: center;
          justify-content: center;
          flex-wrap: wrap;

          max-width: 800px;
          margin-top: 3rem;
        }

        .card {
          margin: 1rem;
          flex-basis: 45%;
          padding: 1.5rem;
          text-align: left;
          color: inherit;
          text-decoration: none;
          border: 1px solid #eaeaea;
          border-radius: 10px;
          transition: color 0.15s ease, border-color 0.15s ease;
        }

        .card:hover,
        .card:focus,
        .card:active {
          color: #004b89;
          border-color: #004b89;
        }

        .card h3 {
          margin: 0 0 1rem 0;
          font-size: 1.5rem;
        }

        .card p {
          margin: 0;
          font-size: 1.25rem;
          line-height: 1.5;
        }

        .logo {
          height: 1em;
        }

        @media (max-width: 600px) {
          .grid {
            width: 100%;
            flex-direction: column;
          }
        }
      \`}</style>

      <style>{\`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: Roboto,
            Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
            sans-serif;
        }

        * {
          box-sizing: border-box;
        }
      \`}</style>
    </div>
  );
}
",
  "ui/public/favicon.ico": "         h     (                                                                                                    v�� v��*v��iv��hv��hv��hv��du���D �K d�K h�K h�K h�K i�K *�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��dv���v���v���v���v���u��9�D 9�K �K ��J �K ��K ��K d�K  u�� u��u��<u��;u��;u��<u��9s���D �J <�PB�yop�X&H�I >�K �K  �D  �D �E >�E @�E @�E @�E ;�D �^1 n��u��^v���u��wr��s��     �K  �K e�K ��K ��K ��K ��K K ?�   v��cv���v��5v���v��|v��v�� �K  �K j�K j�K  �K  �K  �K ��O
Gu��^v���v��\\"v�� v��v���v��yv��	�K  �K j�K j�K  �K  �K  �I ��wjtv���v��7v�� v�� v�� v��v���v��Y�K  �K j�K j�K  �K  �K  �K ��WLu��vv���v��v�� v��
v���v���v���K  �K i�K ��K h�K j�K j�K ��J Ac��v��|v���v��v���v���v��v�� �K  �K *�K j�K j�K j�K j�K e�K �gC v��v��yv���v���v��v��                                         v�� v��v��Xv��v��         ��  �  �  �  �  �  �  �  ��  ��  �  �8  �  �  ��  ��  ",
}
`;

exports[`Test the create-somod with out git eslint and prettier 1`] = `
Object {
  ".gitignore": "node_modules
.somod
/build
/parameters.json
.aws-sam
samconfig.toml
/template.yaml
.next
tsconfig.json
/pages
/public
next-env.d.ts
.env
next.config.js",
  "lib/index.ts": "export const Welcome = (): string => {
  return \\"Welcome to SOMOD Framework. learn more at https://somod.dev\\";
};
",
  "package.json": "{
  \\"name\\": \\"my-module\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"scripts\\": {
    \\"build\\": \\"npx somod build\\",
    \\"pretest\\": \\"npm run build\\",
    \\"test\\": \\"echo 'No Tests'\\",
    \\"prepack\\": \\"npm run test\\",
    \\"start\\": \\"npx somod start\\",
    \\"deploy\\": \\"npx somod deploy\\"
  },
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"ISC\\",
  \\"devDependencies\\": {
    \\"somod\\": \\"^1.15.10\\"
  },
  \\"module\\": \\"build/lib/index.js\\",
  \\"typings\\": \\"build/lib/index.d.ts\\",
  \\"files\\": [
    \\"build\\"
  ],
  \\"sideEffects\\": false,
  \\"somod\\": \\"1.15.10\\"
}
",
  "parameters.yaml": "# yaml-language-server: $schema=./node_modules/somod-schema/schemas/parameters/index.json

parameters: {}
",
  "serverless/template.yaml": "# yaml-language-server: $schema=../node_modules/somod-schema/schemas/serverless-template/index.json

Resources:
  SampleParameter:
    Type: AWS::SSM::Parameter
    Properties:
      Name: Hello
      Description: \\"TODO: This is a sample resource, Delete this and add the valid resources for your module\\"
      Type: String
      Value: Good Luck
",
  "tsconfig.somod.json": "{
  \\"compilerOptions\\": {
    \\"allowUmdGlobalAccess\\": true,
    \\"outDir\\": \\"build\\",
    \\"declaration\\": true,
    \\"target\\": \\"ES5\\",
    \\"module\\": \\"ESNext\\",
    \\"rootDir\\": \\"./\\",
    \\"lib\\": [\\"ESNext\\", \\"DOM\\", \\"DOM.Iterable\\"],
    \\"moduleResolution\\": \\"Node\\",
    \\"esModuleInterop\\": true,
    \\"importHelpers\\": true,
    \\"skipLibCheck\\": true,
    \\"jsx\\": \\"react-jsx\\"
  },
  \\"include\\": [\\"lib\\", \\"ui\\", \\"serverless\\"],
  \\"exclude\\": []
}
",
  "ui/config.yaml": "# yaml-language-server: $schema=../node_modules/somod-schema/schemas/ui-config/index.json

env: {}
imageDomains: []
publicRuntimeConfig: {}
serverRuntimeConfig: {}
",
  "ui/pages/_document.tsx": "import { Html, Head, Main, NextScript } from \\"next/document\\";

const Document = () => {
  return (
    <Html>
      <Head>
        <link
          href=\\"https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@300;400;500;700&family=Roboto:wght@300;400;500;700&display=swap\\"
          rel=\\"stylesheet\\"
        />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
};

export default Document;
",
  "ui/pages/index.tsx": "import Head from \\"next/head\\";

export default function Home() {
  return (
    <div className=\\"container\\">
      <Head>
        <title key=\\"title\\">Create Somod Module</title>
        <link key=\\"favicon\\" rel=\\"icon\\" href=\\"/favicon.ico\\" />
      </Head>

      <main>
        <h1 className=\\"title\\">
          Welcome to <a href=\\"https://somod.dev\\">SOMOD!</a>
        </h1>

        <p className=\\"description\\">
          Get started by editing <code>ui/pages/index.tsx</code>
        </p>

        <div className=\\"grid\\">
          <a href=\\"https://docs.somod.dev\\" className=\\"card\\">
            <h3>Documentation &rarr;</h3>
            <p>Find in-depth information about SOMOD features and API.</p>
          </a>

          <a
            href=\\"https://www.youtube.com/channel/UCl6JNhHf9T7O1mxgnwwXvLw\\"
            className=\\"card\\"
          >
            <h3>Learn &rarr;</h3>
            <p>Learn SOMOD framework from these Explainer videos</p>
          </a>

          <a href=\\"https://github.com/somod-dev\\" className=\\"card\\">
            <h3>Examples &rarr;</h3>
            <p>Discover and deploy boilerplate example SOMOD projects.</p>
          </a>
        </div>
      </main>

      <footer>
        <a href=\\"https://somod.dev\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
          Powered by{\\" \\"}
          <svg
            width=\\"24\\"
            height=\\"24\\"
            viewBox=\\"0 0 100 100\\"
            version=\\"1.1\\"
            xmlns=\\"http://www.w3.org/2000/svg\\"
          >
            <polygon
              fill=\\"#ffb476\\"
              points=\\"0,0 100,0 100,20 40,20 40,50 0,50\\"
            ></polygon>
            <polygon
              fill=\\"#004b89\\"
              points=\\"0,80 60,80 60,50 100,50 100,100 0,100\\"
            ></polygon>
          </svg>
        </a>
      </footer>

      <style>{\`
        .container {
          min-height: 100vh;
          padding: 0 0.5rem;
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
        }

        main {
          padding: 5rem 0;
          flex: 1;
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
        }

        footer {
          width: 100%;
          height: 100px;
          border-top: 1px solid #eaeaea;
          display: flex;
          justify-content: center;
          align-items: center;
        }

        footer svg {
          margin-left: 0.5rem;
        }

        footer a {
          display: flex;
          justify-content: center;
          align-items: center;
        }

        a {
          color: inherit;
          text-decoration: none;
        }

        .title a {
          color: #004b89;
          text-decoration: none;
        }

        .title a:hover,
        .title a:focus,
        .title a:active {
          text-decoration: underline;
        }

        .title {
          margin: 0;
          line-height: 1.15;
          font-size: 4rem;
        }

        .title,
        .description {
          text-align: center;
        }

        .description {
          line-height: 1.5;
          font-size: 1.5rem;
        }

        code {
          background: #fafafa;
          border-radius: 5px;
          padding: 0.75rem;
          font-size: 1.1rem;
          font-family: Menlo, Monaco, Lucida Console, Liberation Mono,
            DejaVu Sans Mono, Bitstream Vera Sans Mono, Courier New, monospace;
        }

        .grid {
          display: flex;
          align-items: center;
          justify-content: center;
          flex-wrap: wrap;

          max-width: 800px;
          margin-top: 3rem;
        }

        .card {
          margin: 1rem;
          flex-basis: 45%;
          padding: 1.5rem;
          text-align: left;
          color: inherit;
          text-decoration: none;
          border: 1px solid #eaeaea;
          border-radius: 10px;
          transition: color 0.15s ease, border-color 0.15s ease;
        }

        .card:hover,
        .card:focus,
        .card:active {
          color: #004b89;
          border-color: #004b89;
        }

        .card h3 {
          margin: 0 0 1rem 0;
          font-size: 1.5rem;
        }

        .card p {
          margin: 0;
          font-size: 1.25rem;
          line-height: 1.5;
        }

        .logo {
          height: 1em;
        }

        @media (max-width: 600px) {
          .grid {
            width: 100%;
            flex-direction: column;
          }
        }
      \`}</style>

      <style>{\`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: Roboto,
            Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
            sans-serif;
        }

        * {
          box-sizing: border-box;
        }
      \`}</style>
    </div>
  );
}
",
  "ui/public/favicon.ico": "         h     (                                                                                                    v�� v��*v��iv��hv��hv��hv��du���D �K d�K h�K h�K h�K i�K *�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��dv���v���v���v���v���u��9�D 9�K �K ��J �K ��K ��K d�K  u�� u��u��<u��;u��;u��<u��9s���D �J <�PB�yop�X&H�I >�K �K  �D  �D �E >�E @�E @�E @�E ;�D �^1 n��u��^v���u��wr��s��     �K  �K e�K ��K ��K ��K ��K K ?�   v��cv���v��5v���v��|v��v�� �K  �K j�K j�K  �K  �K  �K ��O
Gu��^v���v��\\"v�� v��v���v��yv��	�K  �K j�K j�K  �K  �K  �I ��wjtv���v��7v�� v�� v�� v��v���v��Y�K  �K j�K j�K  �K  �K  �K ��WLu��vv���v��v�� v��
v���v���v���K  �K i�K ��K h�K j�K j�K ��J Ac��v��|v���v��v���v���v��v�� �K  �K *�K j�K j�K j�K j�K e�K �gC v��v��yv���v���v��v��                                         v�� v��v��Xv��v��         ��  �  �  �  �  �  �  �  ��  ��  �  �8  �  �  ��  ��  ",
}
`;

exports[`Test the create-somod without options 1`] = `
Object {
  ".eslintignore": "node_modules
.somod
/build
/parameters.json
.aws-sam
samconfig.toml
/template.yaml
.next
tsconfig.json
/pages
/public
next-env.d.ts
.env
next.config.js",
  ".gitignore": "node_modules
.somod
/build
/parameters.json
.aws-sam
samconfig.toml
/template.yaml
.next
tsconfig.json
/pages
/public
next-env.d.ts
.env
next.config.js",
  ".prettierignore": "node_modules
.somod
/build
/parameters.json
.aws-sam
samconfig.toml
/template.yaml
.next
tsconfig.json
/pages
/public
next-env.d.ts
.env
next.config.js",
  "lib/index.ts": "export const Welcome = (): string => {
  return \\"Welcome to SOMOD Framework. learn more at https://somod.dev\\";
};
",
  "package.json": "{
  \\"name\\": \\"my-module\\",
  \\"version\\": \\"1.0.0\\",
  \\"description\\": \\"\\",
  \\"scripts\\": {
    \\"prettier\\": \\"npx prettier --check --ignore-unknown --no-error-on-unmatched-pattern ./**/*\\",
    \\"eslint\\": \\"npx eslint ./ --no-error-on-unmatched-pattern\\",
    \\"prebuild\\": \\"npm run prettier && npm run eslint\\",
    \\"build\\": \\"npx somod build\\",
    \\"pretest\\": \\"npm run build\\",
    \\"test\\": \\"echo 'No Tests'\\",
    \\"prepack\\": \\"npm run test\\",
    \\"start\\": \\"npx somod start\\",
    \\"deploy\\": \\"npx somod deploy\\"
  },
  \\"keywords\\": [],
  \\"author\\": \\"\\",
  \\"license\\": \\"ISC\\",
  \\"devDependencies\\": {
    \\"eslint-config-next\\": \\"^13.1.6\\",
    \\"eslint-config-sodaru\\": \\"^1.0.1\\",
    \\"prettier-config-sodaru\\": \\"^1.0.0\\",
    \\"somod\\": \\"^1.15.10\\"
  },
  \\"module\\": \\"build/lib/index.js\\",
  \\"typings\\": \\"build/lib/index.d.ts\\",
  \\"files\\": [
    \\"build\\"
  ],
  \\"sideEffects\\": false,
  \\"somod\\": \\"1.15.10\\"
}
",
  "parameters.yaml": "# yaml-language-server: $schema=./node_modules/somod-schema/schemas/parameters/index.json

parameters: {}
",
  "serverless/template.yaml": "# yaml-language-server: $schema=../node_modules/somod-schema/schemas/serverless-template/index.json

Resources:
  SampleParameter:
    Type: AWS::SSM::Parameter
    Properties:
      Name: Hello
      Description: \\"TODO: This is a sample resource, Delete this and add the valid resources for your module\\"
      Type: String
      Value: Good Luck
",
  "tsconfig.somod.json": "{
  \\"compilerOptions\\": {
    \\"allowUmdGlobalAccess\\": true,
    \\"outDir\\": \\"build\\",
    \\"declaration\\": true,
    \\"target\\": \\"ES5\\",
    \\"module\\": \\"ESNext\\",
    \\"rootDir\\": \\"./\\",
    \\"lib\\": [\\"ESNext\\", \\"DOM\\", \\"DOM.Iterable\\"],
    \\"moduleResolution\\": \\"Node\\",
    \\"esModuleInterop\\": true,
    \\"importHelpers\\": true,
    \\"skipLibCheck\\": true,
    \\"jsx\\": \\"react-jsx\\"
  },
  \\"include\\": [\\"lib\\", \\"ui\\", \\"serverless\\"],
  \\"exclude\\": []
}
",
  "ui/config.yaml": "# yaml-language-server: $schema=../node_modules/somod-schema/schemas/ui-config/index.json

env: {}
imageDomains: []
publicRuntimeConfig: {}
serverRuntimeConfig: {}
",
  "ui/pages/_document.tsx": "import { Html, Head, Main, NextScript } from \\"next/document\\";

const Document = () => {
  return (
    <Html>
      <Head>
        <link
          href=\\"https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@300;400;500;700&family=Roboto:wght@300;400;500;700&display=swap\\"
          rel=\\"stylesheet\\"
        />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
};

export default Document;
",
  "ui/pages/index.tsx": "import Head from \\"next/head\\";

export default function Home() {
  return (
    <div className=\\"container\\">
      <Head>
        <title key=\\"title\\">Create Somod Module</title>
        <link key=\\"favicon\\" rel=\\"icon\\" href=\\"/favicon.ico\\" />
      </Head>

      <main>
        <h1 className=\\"title\\">
          Welcome to <a href=\\"https://somod.dev\\">SOMOD!</a>
        </h1>

        <p className=\\"description\\">
          Get started by editing <code>ui/pages/index.tsx</code>
        </p>

        <div className=\\"grid\\">
          <a href=\\"https://docs.somod.dev\\" className=\\"card\\">
            <h3>Documentation &rarr;</h3>
            <p>Find in-depth information about SOMOD features and API.</p>
          </a>

          <a
            href=\\"https://www.youtube.com/channel/UCl6JNhHf9T7O1mxgnwwXvLw\\"
            className=\\"card\\"
          >
            <h3>Learn &rarr;</h3>
            <p>Learn SOMOD framework from these Explainer videos</p>
          </a>

          <a href=\\"https://github.com/somod-dev\\" className=\\"card\\">
            <h3>Examples &rarr;</h3>
            <p>Discover and deploy boilerplate example SOMOD projects.</p>
          </a>
        </div>
      </main>

      <footer>
        <a href=\\"https://somod.dev\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
          Powered by{\\" \\"}
          <svg
            width=\\"24\\"
            height=\\"24\\"
            viewBox=\\"0 0 100 100\\"
            version=\\"1.1\\"
            xmlns=\\"http://www.w3.org/2000/svg\\"
          >
            <polygon
              fill=\\"#ffb476\\"
              points=\\"0,0 100,0 100,20 40,20 40,50 0,50\\"
            ></polygon>
            <polygon
              fill=\\"#004b89\\"
              points=\\"0,80 60,80 60,50 100,50 100,100 0,100\\"
            ></polygon>
          </svg>
        </a>
      </footer>

      <style>{\`
        .container {
          min-height: 100vh;
          padding: 0 0.5rem;
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
        }

        main {
          padding: 5rem 0;
          flex: 1;
          display: flex;
          flex-direction: column;
          justify-content: center;
          align-items: center;
        }

        footer {
          width: 100%;
          height: 100px;
          border-top: 1px solid #eaeaea;
          display: flex;
          justify-content: center;
          align-items: center;
        }

        footer svg {
          margin-left: 0.5rem;
        }

        footer a {
          display: flex;
          justify-content: center;
          align-items: center;
        }

        a {
          color: inherit;
          text-decoration: none;
        }

        .title a {
          color: #004b89;
          text-decoration: none;
        }

        .title a:hover,
        .title a:focus,
        .title a:active {
          text-decoration: underline;
        }

        .title {
          margin: 0;
          line-height: 1.15;
          font-size: 4rem;
        }

        .title,
        .description {
          text-align: center;
        }

        .description {
          line-height: 1.5;
          font-size: 1.5rem;
        }

        code {
          background: #fafafa;
          border-radius: 5px;
          padding: 0.75rem;
          font-size: 1.1rem;
          font-family: Menlo, Monaco, Lucida Console, Liberation Mono,
            DejaVu Sans Mono, Bitstream Vera Sans Mono, Courier New, monospace;
        }

        .grid {
          display: flex;
          align-items: center;
          justify-content: center;
          flex-wrap: wrap;

          max-width: 800px;
          margin-top: 3rem;
        }

        .card {
          margin: 1rem;
          flex-basis: 45%;
          padding: 1.5rem;
          text-align: left;
          color: inherit;
          text-decoration: none;
          border: 1px solid #eaeaea;
          border-radius: 10px;
          transition: color 0.15s ease, border-color 0.15s ease;
        }

        .card:hover,
        .card:focus,
        .card:active {
          color: #004b89;
          border-color: #004b89;
        }

        .card h3 {
          margin: 0 0 1rem 0;
          font-size: 1.5rem;
        }

        .card p {
          margin: 0;
          font-size: 1.25rem;
          line-height: 1.5;
        }

        .logo {
          height: 1em;
        }

        @media (max-width: 600px) {
          .grid {
            width: 100%;
            flex-direction: column;
          }
        }
      \`}</style>

      <style>{\`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: Roboto,
            Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue,
            sans-serif;
        }

        * {
          box-sizing: border-box;
        }
      \`}</style>
    </div>
  );
}
",
  "ui/public/favicon.ico": "         h     (                                                                                                    v�� v��*v��iv��hv��hv��hv��du���D �K d�K h�K h�K h�K i�K *�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��hv���v���v���v���v���u��;�D ;�K ��K ��K ��K ��K ��K h�K  v�� v��dv���v���v���v���v���u��9�D 9�K �K ��J �K ��K ��K d�K  u�� u��u��<u��;u��;u��<u��9s���D �J <�PB�yop�X&H�I >�K �K  �D  �D �E >�E @�E @�E @�E ;�D �^1 n��u��^v���u��wr��s��     �K  �K e�K ��K ��K ��K ��K K ?�   v��cv���v��5v���v��|v��v�� �K  �K j�K j�K  �K  �K  �K ��O
Gu��^v���v��\\"v�� v��v���v��yv��	�K  �K j�K j�K  �K  �K  �I ��wjtv���v��7v�� v�� v�� v��v���v��Y�K  �K j�K j�K  �K  �K  �K ��WLu��vv���v��v�� v��
v���v���v���K  �K i�K ��K h�K j�K j�K ��J Ac��v��|v���v��v���v���v��v�� �K  �K *�K j�K j�K j�K j�K e�K �gC v��v��yv���v���v��v��                                         v�� v��v��Xv��v��         ��  �  �  �  �  �  �  �  ��  ��  �  �8  �  �  ��  ��  ",
}
`;
